#163423238 Implement get one party

haywhyze opened the pull request:
#### What does this PR do?
Implements get one party endpoints with accompanying tests

#### Description of Task to be completed?
- write test scripts to test the getOne party functionality
- create a middleware to validate party ID
- create getOne method to handle get request to `/parties/<party-id>`
- add the getOne method and the validateID middleware to the router

#### How should this be manually tested?
- clone this repository by executing `git clone https://github.com/haywhyze/politico.git`
- cd to the cloned local repository `cd politico`
- run `ft-get-one-party-163423238`
- install all dependencies by running `npm install`
- start the server with `npm start`
- when the server is up, you can test using Postman or any other related service.

#### What are the relevant pivotal tracker stories?
[#163423238](https://www.pivotaltracker.com/story/show/163423238)

haywhyze committed:
feat(server): implement get one party

- write test scripts to test the getOne party functionalilty
- create a middleware to validate party ID
- create getOne method to handle get request to `/parties/<party-id>`
- add the getOne method and the validateID middleware to the router

[delivers #163423238]

haywhyze committed:
Update Procfile

- update procfile to make app deployable for easy review

[delivers #163423238]

megame24 commented on server/tests/party.test.js:
```diff
@@ -155,4 +155,37 @@ describe('Parties', () => {
+        .request(app)
+        .get('/api/v1/parties/15')
+        .end((err, res) => {
+          expect(res.status).to.equal(404);
```
you might also want to test that the right error message is returned. Adopt this going forward, and update your tests to test for more than just the status code.

haywhyze replied:
Alright. Thanks.

haywhyze replied:
On this branch, should I update all my tests or just the one that affects features on this PR.

megame24 commented on server/middlewares/validateID.js:
```diff
@@ -0,0 +1,21 @@
+
+const validateID = (req, res, next) => {
+  const id = Number(req.params.id);
+  if (Number.isNaN(id) || id % 1 !== 0 || id < 1 || String(req.params.id).indexOf('.') !== -1) {
```
I don't think that the last 2 checks are necessary since they are basically checking for floating point numbers which has already been taken care of by the second check. What do you think?

haywhyze replied:
The second to the last check is not necessary as you pointed out, but the last check is actually checking for 1.000... or should that be valid?

megame24 replied:
Since you are already converting to absoulte number using `Number(id)`, that last check is not necessary because `Number()` will remove trailing `.000...`

haywhyze replied:
Alright. Thanks

haywhyze committed:
feat(server): implement get one party

- remove unnecessary extra check on party id.
- update get one party's tests scripts to test for the right error message.

[delivers #163423238]

megame24 reviewed (approved):
You can merge this.
